### PR TITLE
Update ffmpeg/ffpyplayer versions

### PR DESCRIPTION
**Pull request by tito**

### CODE REVIEW

**Review by AndreMiras — APPROVED** [review]

File: kivy_ios/recipes/ffpyplayer/__init__.py
Comment (misl6) [inline kivy_ios/recipes/ffpyplayer/__init__.py:9]: Hi @tito!

Seems that the CI job "build_updated_recipes" is failing as we missed a needed dependency in `depends` list,
This will likely go unnoticed while building it manually but is definitely something nice to fix.

I've added a suggestion here:
  
```suggestion
    depends = ["python", "sdl2", "ffmpeg"]
```

**Comment by misl6** [comment]
@tito 

Are these changes also fixing the `ccache` related issue we talked about on Discord?

File: kivy_ios/recipes/ffpyplayer/__init__.py
Comment (AndreMiras) [inline kivy_ios/recipes/ffpyplayer/__init__.py:7]: I slightly overlooked that change, but don't we usually exclude the `v` from the version so the package pinning won't have to include it?
So I mean `url` could be:
```python
url = "https://github.com/matham/ffpyplayer/archive/v{version}.zip"
```

Comment (tito) [reply kivy_ios/recipes/ffpyplayer/__init__.py:7]: Yes/no ? Otherwise there is no way to go back to a specific archive if not pinned with a v ?

Comment (misl6) [reply kivy_ios/recipes/ffpyplayer/__init__.py:7]: 🤔 That's a good question.

Pros:
- Users will likely try to pin versions like `ffpyplayer==4.x.x`, as they are used to doing the same when installing things via pip (and etc).

Cons:
- We can't pin a specific hash or tag (if that tag is not prepended by `v`) without changing the URL

Some personal considerations about it + one extra question:
- Recipes are not expected to work fine between multiple versions of the same package (or at least are not tested), so pinning a specific version of a package that needs a recipe to install should be discouraged IMHO.
- Maybe `ffpyplayer` is not the case, but sometimes we need to target a specific tag or hash as a temp workaround (that happened in past with SDL2 and Kivy).
- Maybe we should print a **WARNING** regarding potential unexpected issues while pinning a specific version on a recipe? 

BTW, I'm +1 for keeping the `v` into the version.

Comment (AndreMiras) [reply kivy_ios/recipes/ffpyplayer/__init__.py:7]: All good point thanks guys, I leave that to you, I'm fine either ways :+1:

**Comment by misl6** [comment]
@tito If you're good with the suggestion and then the CI flags it as ✅ , I'm fine with it.

**Comment by brentpicasso** [comment]
Hi, any reason this cannot be merged? 

Thanks!

**Comment by AndreMiras** [comment]
> Hi, any reason this cannot be merged?

Thanks for the heads up, I've rebased the PR, will merge if the CI is OK

**Comment by AndreMiras** [comment]
The `build_updated_recipes` failed, the error was `fatal error: 'SDL_version.h' file not found`.
Here's an extract with more context, for full context see the build.
```
2023-03-07T10:12:41.8427530Z [DEBUG   ] /var/folders/24/8k48jl6d249_n_qfxwsl6xvm0000gn/T/tmpv1azr2cn -I/Users/runner/work/kivy-ios/kivy-ios/dist/include/x86_64/libffi -O3 -miphoneos-version-min=9.0 -I/Users/runner/work/kivy-ios/kivy-ios/dist/include/x86_64/ffmpeg -I/Users/runner/work/kivy-ios/kivy-ios/dist/include/x86_64/ffi -I/Users/runner/work/kivy-ios/kivy-ios/dist/include/x86_64/openssl -I/Users/runner/work/kivy-ios/kivy-ios/dist/include/x86_64 -g -I/Users/runner/work/kivy-ios/kivy-ios/dist/include/common/sdl2 -I/Users/runner/work/kivy-ios/kivy-ios/dist/include/x86_64/ffmpeg -I/Users/runner/work/kivy-ios/kivy-ios/build/ffpyplayer/x86_64/ffpyplayer-4.3.5/ffpyplayer -I/Users/runner/work/kivy-ios/kivy-ios/build/ffpyplayer/x86_64/ffpyplayer-4.3.5/ffpyplayer/includes -I/Users/runner/work/kivy-ios/kivy-ios/dist/hostpython3/include/python3.10 -c /Users/runner/work/kivy-ios/kivy-ios/build/ffpyplayer/x86_64/ffpyplayer-4.3.5/ffpyplayer/clib/misc.c -o build/temp.macosx-12.6-x86_64-cpython-310/Users/runner/work/kivy-ios/kivy-ios/build/ffpyplayer/x86_64/ffpyplayer-4.3.5/ffpyplayer/clib/misc.o -O3 -fno-strict-aliasing -Wno-error
2023-03-07T10:12:41.8992820Z [DEBUG   ] In file included from /Users/runner/work/kivy-ios/kivy-ios/build/ffpyplayer/x86_64/ffpyplayer-4.3.5/ffpyplayer/clib/misc.c:1:
2023-03-07T10:12:41.8994170Z [DEBUG   ] In file included from /Users/runner/work/kivy-ios/kivy-ios/build/ffpyplayer/x86_64/ffpyplayer-4.3.5/ffpyplayer/clib/misc.h:5:
2023-03-07T10:12:41.8995440Z [DEBUG   ] /Users/runner/work/kivy-ios/kivy-ios/build/ffpyplayer/x86_64/ffpyplayer-4.3.5/ffpyplayer/clib/../includes/ffconfig.h:5:10: fatal error: 'SDL_version.h' file not found
2023-03-07T10:12:41.8996250Z [DEBUG   ] #include "SDL_version.h"
2023-03-07T10:12:41.8996870Z [DEBUG   ]          ^~~~~~~~~~~~~~~
2023-03-07T10:12:41.9605940Z [DEBUG   ] 1 error generated.
2023-03-07T10:12:41.9707650Z [DEBUG   ] error: command '/var/folders/24/8k48jl6d249_n_qfxwsl6xvm0000gn/T/tmpv1azr2cn' failed with exit code 1
2023-03-07T10:12:42.0437480Z Exception in thread background thread for pid 93873:
2023-03-07T10:12:42.0439300Z Traceback (most recent call last):
2023-03-07T10:12:42.0440370Z   File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/threading.py", line 1038, in _bootstrap_inner
2023-03-07T10:12:42.0442150Z     self.run()
2023-03-07T10:12:42.0443830Z   File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/threading.py", line 975, in run
2023-03-07T10:12:42.0445580Z     self._target(*self._args, **self._kwargs)
2023-03-07T10:12:42.0447020Z   File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/sh.py", line 1540, in wrap
2023-03-07T10:12:42.0457140Z     fn(*args, **kwargs)
2023-03-07T10:12:42.0458870Z   File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/sh.py", line 2459, in background_thread
2023-03-07T10:12:42.0459660Z     handle_exit_code(exit_code)
2023-03-07T10:12:42.0460740Z   File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/sh.py", line 2157, in fn
2023-03-07T10:12:42.0466960Z     return self.command.handle_command_exit_code(exit_code)
2023-03-07T10:12:42.0467690Z            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2023-03-07T10:12:42.0468740Z   File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/sh.py", line 815, in handle_command_exit_code
2023-03-07T10:12:42.0469600Z     raise exc
2023-03-07T10:12:42.0470190Z sh.ErrorReturnCode_1: 
2023-03-07T10:12:42.0470670Z 
2023-03-07T10:12:42.0471370Z   RAN: /Users/runner/work/kivy-ios/kivy-ios/dist/hostpython3/bin/python setup.py build_ext -g
```
@brentpicasso do you have a working version of the recipe?

**Comment by brentpicasso** [comment]
Thanks for checking it.  I don't have this building locally.  

@tito  - were you able to get it to build at the time you submitted this PR?

**Comment by misl6** [comment]
> The `build_updated_recipes` failed, the error was `fatal error: 'SDL_version.h' file not found`. Here's an extract with more context, for full context see the build.
> 
> ```
> 2023-03-07T10:12:41.8427530Z [DEBUG   ] /var/folders/24/8k48jl6d249_n_qfxwsl6xvm0000gn/T/tmpv1azr2cn -I/Users/runner/work/kivy-ios/kivy-ios/dist/include/x86_64/libffi -O3 -miphoneos-version-min=9.0 -I/Users/runner/work/kivy-ios/kivy-ios/dist/include/x86_64/ffmpeg -I/Users/runner/work/kivy-ios/kivy-ios/dist/include/x86_64/ffi -I/Users/runner/work/kivy-ios/kivy-ios/dist/include/x86_64/openssl -I/Users/runner/work/kivy-ios/kivy-ios/dist/include/x86_64 -g -I/Users/runner/work/kivy-ios/kivy-ios/dist/include/common/sdl2 -I/Users/runner/work/kivy-ios/kivy-ios/dist/include/x86_64/ffmpeg -I/Users/runner/work/kivy-ios/kivy-ios/build/ffpyplayer/x86_64/ffpyplayer-4.3.5/ffpyplayer -I/Users/runner/work/kivy-ios/kivy-ios/build/ffpyplayer/x86_64/ffpyplayer-4.3.5/ffpyplayer/includes -I/Users/runner/work/kivy-ios/kivy-ios/dist/hostpython3/include/python3.10 -c /Users/runner/work/kivy-ios/kivy-ios/build/ffpyplayer/x86_64/ffpyplayer-4.3.5/ffpyplayer/clib/misc.c -o build/temp.macosx-12.6-x86_64-cpython-310/Users/runner/work/kivy-ios/kivy-ios/build/ffpyplayer/x86_64/ffpyplayer-4.3.5/ffpyplayer/clib/misc.o -O3 -fno-strict-aliasing -Wno-error
> 2023-03-07T10:12:41.8992820Z [DEBUG   ] In file included from /Users/runner/work/kivy-ios/kivy-ios/build/ffpyplayer/x86_64/ffpyplayer-4.3.5/ffpyplayer/clib/misc.c:1:
> 2023-03-07T10:12:41.8994170Z [DEBUG   ] In file included from /Users/runner/work/kivy-ios/kivy-ios/build/ffpyplayer/x86_64/ffpyplayer-4.3.5/ffpyplayer/clib/misc.h:5:
> 2023-03-07T10:12:41.8995440Z [DEBUG   ] /Users/runner/work/kivy-ios/kivy-ios/build/ffpyplayer/x86_64/ffpyplayer-4.3.5/ffpyplayer/clib/../includes/ffconfig.h:5:10: fatal error: 'SDL_version.h' file not found
> 2023-03-07T10:12:41.8996250Z [DEBUG   ] #include "SDL_version.h"
> 2023-03-07T10:12:41.8996870Z [DEBUG   ]          ^~~~~~~~~~~~~~~
> 2023-03-07T10:12:41.9605940Z [DEBUG   ] 1 error generated.
> 2023-03-07T10:12:41.9707650Z [DEBUG   ] error: command '/var/folders/24/8k48jl6d249_n_qfxwsl6xvm0000gn/T/tmpv1azr2cn' failed with exit code 1
> 2023-03-07T10:12:42.0437480Z Exception in thread background thread for pid 93873:
> 2023-03-07T10:12:42.0439300Z Traceback (most recent call last):
> 2023-03-07T10:12:42.0440370Z   File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/threading.py", line 1038, in _bootstrap_inner
> 2023-03-07T10:12:42.0442150Z     self.run()
> 2023-03-07T10:12:42.0443830Z   File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/threading.py", line 975, in run
> 2023-03-07T10:12:42.0445580Z     self._target(*self._args, **self._kwargs)
> 2023-03-07T10:12:42.0447020Z   File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/sh.py", line 1540, in wrap
> 2023-03-07T10:12:42.0457140Z     fn(*args, **kwargs)
> 2023-03-07T10:12:42.0458870Z   File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/sh.py", line 2459, in background_thread
> 2023-03-07T10:12:42.0459660Z     handle_exit_code(exit_code)
> 2023-03-07T10:12:42.0460740Z   File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/sh.py", line 2157, in fn
> 2023-03-07T10:12:42.0466960Z     return self.command.handle_command_exit_code(exit_code)
> 2023-03-07T10:12:42.0467690Z            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
> 2023-03-07T10:12:42.0468740Z   File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/sh.py", line 815, in handle_command_exit_code
> 2023-03-07T10:12:42.0469600Z     raise exc
> 2023-03-07T10:12:42.0470190Z sh.ErrorReturnCode_1: 
> 2023-03-07T10:12:42.0470670Z 
> 2023-03-07T10:12:42.0471370Z   RAN: /Users/runner/work/kivy-ios/kivy-ios/dist/hostpython3/bin/python setup.py build_ext -g
> ```
> 
> @brentpicasso do you have a working version of the recipe?

Hi @AndreMiras !

See: https://github.com/kivy/kivy-ios/pull/716#discussion_r895145331

**Comment by brentpicasso** [comment]
Thanks @AndreMiras and @misl6 for the quick turnaround on this!

**Review by AndreMiras — APPROVED** [review]
Thanks tito and misl6, merging

**Pull request merged**
